plugins/binanceExchange_ws.go

```diff
@@ -398,6 +397,12 @@ func (beWs *binanceExchangeWs) GetOrderBook(pair *model.TradingPair, maxCount in
 	askCcxtOrders := book.Asks
 	bidCcxtOrders := book.Bids
 
+	if fetchLimit != 20 {
+		// we may not have fetched all the requested levels because the exchange may not have had that many levels in depth
+		askCcxtOrders = askCcxtOrders[:fetchLimit]
```

Comment:
what is fetchLimit is 20 and we only fetched 4 values? would that give a `runtime error: slice bounds out of range` error?

the right thing to do would be to do what you had earlier and remove the `fetchLimit != 20` check. can we do that please? 

see previous comment about suggested code: https://github.com/stellar/kelp/pull/718#discussion_r671862467